hal/common/file.hpp

```diff
@@ -142,14 +142,21 @@ class File
 	 *
 	 * \return 	success	(always false)
 	 */
-	 bool flush();
+	bool flush();
 
 	/**
 	 * \brief 	write newline character to file ('\n')
 	 *
 	 * \return 	success
 	 */
-	 bool newline();
+	bool newline();
+
+	/**
+	 * \brief	sync the file
```

Comment:
sync is in fact only needed in fat_fs. 
In fat_fs, when you do a write operation, it do not write to the file but rather to a temporary memory, it is only when you do f_sync, f_close (or f_stat, but this one I'm not sure) that the data is really written to the file. 
I can use the flush function, but it does not make much sense to link flush with sync in my opinion.